contracts/nexus-gateway/src/msg.rs

```diff
@@ -9,6 +11,6 @@ pub struct InstantiateMsg {
 
 #[cw_serde]
 pub enum ExecuteMsg {
-    VerifyMessages(Vec<Message>),
-    RouteMessages(Vec<Message>),
+    RouteMessagesToNexus(Vec<Message>),
+    RouteMessagesToRouter(Vec<nexus::Message>),
```

Comment:
There is a good reason here. We don't want the nexus module to have to receive and send different things from and to this nexus gateway. Gateway interface consistency doesn't matter when nothing external ever interacts with it.

Reply:
why can't the nexus module just receive and send `connection_router::state::Message` via `RouteMessages`?

Reply:
Cuz again, we need to avoid core having to make any change due to the wasm contracts, i.e. whatever change occurs to the `connection_router::state::Message` (rename of fields, field removal, new field added) should not affect core.

Reply:
Those things are protocol changes, that I don't think would happen very often, and would need to be backwards compatible anyways, else we will break the entire protocol.

At the very least, I think the connection router still should just call `RouteMessages` when passing a message to the nexus gateway, to keep it consistent with every other gateway. This really is just a gateway. So `RouteMessagesToNexus` should just stay as `RouteMessages`. And I would just rename `RouteMessagesToRouter` to something like `TransformAndRouteMessages`.

Reply:
Often or not, protocol changes in wash should have minimum impact on core. Sure I can keep it as RouteMessages for going to nexus. Don't see why the other one should have that "transform" though.

Reply:
Actually, I may have an idea. Let me try some.

Reply:
renamed